### PR TITLE
Fixing https://github.com/wso2/product-is/issues/1951

### PR DESCRIPTION
**Pull request by SachithUd**
### Proposed changes in this pull request

[List all changes you want to add here. If you fixed an issue, please
add a reference to that issue as well.]

-

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminService.java [hunk 1]
```diff
@@ -619,18 +622,50 @@ public void removeOAuthApplicationData(String consumerKey) throws IdentityOAuthA
...
+            OAuthRevocationRequestDTO oAuthRevocationRequestDTO = new OAuthRevocationRequestDTO();
+            oAuthRevocationRequestDTO.setApps(new String[]{oAuthConsumerAppDTO.getApplicationName()});
+            oAuthRevocationRequestDTO.setAuthzUser(username);
+            oAuthRevocationRequestDTO.setConsumerKey(oAuthConsumerAppDTO.getOauthConsumerKey());
+            oAuthRevocationRequestDTO.setConsumerSecret(oAuthConsumerAppDTO.getOauthConsumerSecret());
+            if (log.isDebugEnabled()) {
+                log.debug("Revoking access tokens for Client ID: " + oAuthConsumerAppDTO.getOauthConsumerKey() +
+                        " & user: " + username);
+            }
+            return revokeAuthzForApps(oAuthRevocationRequestDTO);
```
Comment (thanujalk) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminService.java]: is this return statement location is correct? or do we need the for loop?

Comment (GayanM) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminService.java]: +1 need to fix

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminService.java [hunk 2]
```diff
@@ -619,18 +622,53 @@ public void removeOAuthApplicationData(String consumerKey) throws IdentityOAuthA
...
+    public OAuthConsumerAppDTO[] getAppsAuthorizedByUser() throws IdentityOAuthAdminException {
+
+        return getoAuthConsumerAppDTOS(PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername());
+    }
+
+    public OAuthRevocationResponseDTO[] revokeAuthorization(String username) throws IdentityOAuthAdminException {
+
+        OAuthConsumerAppDTO oAuthConsumerAppDTOs[] = getAuthorizedApps(username);
+        OAuthRevocationResponseDTO oAuthRevocationResponseDTO[] = new
+                OAuthRevocationResponseDTO[oAuthConsumerAppDTOs.length];
```
Comment (thanujalk) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminService.java]: Shall we define oAuthRevocationResponseDTO as a list and when return converts it to array.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminService.java [hunk 3]
```diff
@@ -619,18 +622,53 @@ public void removeOAuthApplicationData(String consumerKey) throws IdentityOAuthA
...
+    /**
+     * Revoke authorization for OAuth apps by admin user
      *
      * @param revokeRequestDTO DTO representing authorized user and apps[]
      * @return revokeRespDTO DTO representing success or failure message
      */
-    public OAuthRevocationResponseDTO revokeAuthzForAppsByResoureOwner(
-            OAuthRevocationRequestDTO revokeRequestDTO) throws IdentityOAuthAdminException {
+    public OAuthRevocationResponseDTO revokeAuthzForApps (OAuthRevocationRequestDTO revokeRequestDTO) throws
+            IdentityOAuthAdminException {
```
Comment (thanujalk) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminService.java]: Add a empty line

**Comment by denuwanthi** [comment]
Tested this with latest 5.4 pack.
Revokation of access token not happening.

Steps:
1.create user denuwanthi in IS.
2.Create a service provider called palyground and configure oauth2 as inbound authentication
3.Generate a token for the user denuwanthi.
4.Invoke the revokeAuthorization method as follows

<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsd="http://org.apache.axis2/xsd">
   <soapenv:Header/>
   <soapenv:Body>
      <xsd:revokeAuthorization>
         
         <xsd:username>denuwanthi</xsd:username>
      </xsd:revokeAuthorization>
   </soapenv:Body>
</soapenv:Envelope>

5.Get following error response
<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
   <soapenv:Body>
      <ns:revokeAuthorizationResponse xmlns:ns="http://org.apache.axis2/xsd" xmlns:ax2366="http://base.identity.carbon.wso2.org/xsd" xmlns:ax2369="http://dto.oauth.identity.carbon.wso2.org/xsd" xmlns:ax2365="http://oauth.identity.carbon.wso2.org/xsd">
         <ns:return xsi:type="ax2369:OAuthRevocationResponseDTO" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
            <ax2369:error>false</ax2369:error>
            <ax2369:errorCode xsi:nil="true"/>
            <ax2369:errorMsg xsi:nil="true"/>
         </ns:return>
      </ns:revokeAuthorizationResponse>
   </soapenv:Body>
</soapenv:Envelope>

Back end does not print any error stack trace.

@SachithUd can you confirm this is working properly?

**Comment by pulasthi7** [comment]
@SachithUd Can you look into the issue reported by denuwanthi? Please get it fixed asap.

**Comment by thanujalk** [comment]
Fix is incomplete.

**Pull request closed without merging**